src/read.c

```diff
@@ -1481,6 +1481,23 @@ static avifBool avifParseItemPropertyAssociation(avifMeta * meta, const uint8_t
                 }
             }
             if (supportedType) {
+                if (!essential) {
+                    // Verify that it is legal for this property to not be flagged as essential. Any
+                    // types in this list are *required* in the spec to be flagged as essential when
+                    // associated with an item.
+                    static const char * essentialTypes[] = {
+                        "av1C" // AVIF: Section 2.2.1: "This property shall be marked as essential."
```

Comment:
We can add the following:

`"lsel",  // HEIF: Section 6.5.11.1: "essential shall be equal to 1 for an 'lsel' item property."`

Also note the following from MIAF, which I don't know how to interpret:

> 7.3.9 Transformations and derived items
> The image grid, overlay, and identity derivations shall be supported by the MIAF reader.
> All transformative properties associated with coded and derived images required or conditionally required by this document shall be marked as essential, and shall be from the set that are permitted by this document or the applicable profile. No other essential transformative property shall be associated with such images.